src/app/modules/gb-analysis-module/fractalis-components/gb-fractalis-chart/gb-fractalis-chart.component.ts

```diff
@@ -1,37 +1,128 @@
-import {Component, Input, OnInit} from '@angular/core';
+/**
+ * Copyright 2017 - 2018  The Hyve B.V.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import {AfterViewInit, Component, Input} from '@angular/core';
 import {Chart} from '../../../../models/chart-models/chart';
 import {ChartType} from '../../../../models/chart-models/chart-type';
 import {FractalisService} from '../../../../services/fractalis.service';
+import {FractalisChartDescription} from '../../../../models/fractalis-models/fractalis-chart-description';
+import {FractalisData} from '../../../../models/fractalis-models/fractalis-data';
 import {Concept} from '../../../../models/constraint-models/concept';
+import {ChartValidator} from '../../../../utilities/chart-validator';
 
 @Component({
   selector: 'gb-fractalis-chart',
   templateUrl: './gb-fractalis-chart.component.html',
   styleUrls: ['./gb-fractalis-chart.component.css']
 })
-export class GbFractalisChartComponent implements OnInit {
+export class GbFractalisChartComponent implements AfterViewInit {
 
   @Input() chart: Chart;
+  private fractalisChartDescription: FractalisChartDescription;
+  private fractalisChart: object;
```

Comment:
why is fractalisChart an object instead of  more concrete Class like 'FractalisChart'?

Reply:
fractalisChart is something that is retrieved from fractalis, when setting the chart and then it is passed to fractalis, when updating the chart or getting it's parameters, but it is never used directly and GB doesn't really needs to know what is inside. It is a bit weird structure, like `F` from the fractalis-service (which has `any` type). Do you think it would be useful to add a model class for it?

Reply:
I see, maybe give it a more 'black-box'ish name, like, fractalisChartObj or sth. and add a comment